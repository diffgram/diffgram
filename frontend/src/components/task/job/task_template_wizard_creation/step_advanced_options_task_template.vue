<template>
  <v-container fluid data-cy="task-template-advanced-options-step" v-if="job">
    <div class="d-flex mb-8 justify-space-between">
      <h1 class="font-weight-medium text--primary mr-4" data-cy="task-template-advanced-options-step-title">
        Advanced Options:
      </h1>
    </div>

    <v_error_multiple :error="error"></v_error_multiple>
    <p class="text--primary">
      Optional Configurations.
    </p>


    <v-container fluid>

      <userscript_select
        data-cy="userscrips-select"
        :project_string_id="project_string_id"
        @change="job.default_userscript_id = $event.id"
        label="Choose a Default Userscript"
      >
      </userscript_select>

      <diffgram_select
        :item_list="file_handling_list"
        data-cy="file-handling-select"
        v-model="job.file_handling"
        label="File Handling"
        :disabled="loading"
      >
      </diffgram_select>

    </v-container>

    <wizard_navigation
      @next="on_next_button_click"
      :next_visible="true"
      :loading_next="loading_steps"
      :disabled_next="loading_steps"
      @back="$emit('previous_step')"
      :skip_visible="true">

    </wizard_navigation>
  </v-container>

</template>

<script lang="ts">

  import guide_selector from '../../guide/guide_selector.vue'
  import userscript_select from '../../../annotation/image_and_video_annotation/userscript/userscript_select.vue'
  import Vue from "vue";

  export default Vue.extend({
      name: 'step_guides_task_template',
      props: [
        'project_string_id',
        'job',
        'loading_steps',
      ],

      components: {
        guide_selector,
        userscript_select,
      },

      data() {
        return {
          error: {},
          show_credentials: false,
          loading: false,
          type_list: [
            {
            display_name: 'Normal',
            value: 'Normal'
            },
            {
              display_name: 'Exam',
              value: 'exam_template'
            }
          ],
          share_list: [],
          file_handling_list: [
            {
              'display_name': 'Use Existing (Default)',
              'name': 'use_existing',
              'icon': 'mdi-cached',
              'color': 'primary'
            },
            {
              'display_name': 'Isolate (New Versions of Files)',
              'name': 'isolate',
              'icon': 'mdi-ab-testing',
              'color': 'primary'
            }
          ],
        }
      },
      created() {

      },

      methods: {
        on_next_button_click: function () {
          this.$emit('next_step');
        }
      }
    }
  ) </script>
