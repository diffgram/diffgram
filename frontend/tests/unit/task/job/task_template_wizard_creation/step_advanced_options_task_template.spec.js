import {shallowMount, createLocalVue} from "@vue/test-utils";
import Vuex from 'vuex'
import step_advanced_options_task_template
  from "../../../../../src/components/task/job/task_template_wizard_creation/step_advanced_options_task_template";
import VueRouter from 'vue-router'

const localVue = createLocalVue();
localVue.use(Vuex)
localVue.use(VueRouter)
const router = new VueRouter()

let job = {
  name: 'test',
  label_mode: 'closed_all_available',

  loading: false,
  passes_per_file: 1,
  share_object: {
    // TODO this may fail for org jobs? double check this.
    'text': String,
    'type': 'project'
  },
  share: 'project',
  instance_type: 'box', //"box" or "polygon" or... "text"...
  permission: 'all_secure_users',
  field: 'Other',
  category: 'visual',
  attached_directories_dict: {attached_directories_list: []},
  type: 'Normal',
  connector_data: {},
  // default to no review while improving review system
  review_by_human_freqeuncy: 'No review', //'every_3rd_pass'
  td_api_trainer_basic_training: false,
  file_handling: "use_existing",
  interface_connection: undefined,
  member_list_ids: ["all"]
}
describe("step_advanced_options_task_template.vue", () => {
  let actions;
  let store;

  beforeEach(() => {
    actions = {
      actionClick: jest.fn(),
      actionInput: jest.fn()
    }
    store = new Vuex.Store({
      actions
    })
  });


  it("Renders component correctly.", () => {
    const wrapper = shallowMount(step_advanced_options_task_template, {
      store,
      localVue,
      propsData: {
        job: job
      }
    });
    expect(wrapper).toMatchSnapshot();
  });



  it("Emits event when calling on_next_button_click()", () => {
    const wrapper = shallowMount(step_advanced_options_task_template, {
      store,
      localVue,
      propsData: {
        job: job
      }
    });

    wrapper.vm.on_next_button_click()
    expect(wrapper.emitted().next_step.length).toBe(1);
  });



});
