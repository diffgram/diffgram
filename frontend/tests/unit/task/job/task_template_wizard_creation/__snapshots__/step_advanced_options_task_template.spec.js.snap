// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`step_advanced_options_task_template.vue Renders component correctly. 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
